Microsoft.Azure.Cosmos/src/Telemetry/OpenTelemetry/Filters/DiagnosticsFilterHelper.cs

```diff
@@ -5,23 +5,32 @@
 namespace Microsoft.Azure.Cosmos.Telemetry.Diagnostics
 {
     using System;
-    using System.Net;
+    using Documents;
 
     internal static class DiagnosticsFilterHelper
     {
-        private static readonly TimeSpan latencyThresholdInMs = TimeSpan.FromMilliseconds(250);
-
         /// <summary>
         /// Allow only when either of below is <b>True</b><br></br>
-        /// 1) Latency is not more than 100 ms<br></br>
+        /// 1) Latency is not more than 100/250 (query) ms<br></br>
         /// 3) HTTP status code is not Success<br></br>
         /// </summary>
         /// <returns>true or false</returns>
-        public static bool IsAllowed(
-            TimeSpan latency, 
-            HttpStatusCode statuscode)
+        public static bool IsTracingNeeded(
+            DistributedTracingOptions config,
+            OpenTelemetryAttributes response)
         {
-            return latency > DiagnosticsFilterHelper.latencyThresholdInMs || !statuscode.IsSuccess();
+            TimeSpan latencyThreshold;
+
+            if (config != null && config.DiagnosticsLatencyThreshold.HasValue)
+            {
+                latencyThreshold = config.DiagnosticsLatencyThreshold.Value;
+            }
+            else
+            {
+                latencyThreshold = response.OperationType == OperationType.Query ? DistributedTracingOptions.DefaultQueryTimeoutThreshold : DistributedTracingOptions.DefaultCrudLatencyThreshold;
+            }
+
+            return !(response.Diagnostics.GetClientElapsedTime() < latencyThreshold && response.StatusCode.IsSuccess());
```

Comment:
I'm a bit slow so the negation made me reel back a bit to understand. Would it be clearer without the negation? 

```suggestion
            return response.Diagnostics.GetClientElapsedTime() > latencyThreshold || !response.StatusCode.IsSuccess();
```